demo/src/app/components/datepicker/demos/keyboard/datepicker-keyboard.ts

```diff
@@ -10,14 +10,14 @@ const Key = {
 
 @Injectable()
 export class CustomKeyboardService extends NgbDatepickerKeyboardService {
-  processKey(event: KeyboardEvent, dp: NgbDatepicker, calendar: NgbCalendar) {
+  processKey(event: KeyboardEvent, dp: NgbDatepicker) {
```

Comment:
Can't remove `calendar: NgbCalendar` here, because`dp.calendar` is not a public API

Reply:
Generally, each datepicker instance is tied with a specific calendar instance. So maybe it's better to export the calendar instance via a getter.